### PR TITLE
New files module ported from streams-contrib

### PR DESCRIPTION
**Pull request by johanandren**
Refs #11 & #12

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -14,6 +14,13 @@ lazy val amqp = project
     Dependencies.Amqp
   )
 
+lazy val `akka-stream-alpakka-file` = project
```
Comment (johanandren) [inline build.sbt]: this is a bit ugly but `file` is already used in the scope

File: docs/src/main/paradox/connectors.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 @@@ index
 
+* [File Connectors](file.md)
```
Comment (patriknw) [inline docs/src/main/paradox/connectors.md]: alphabetic order of this list (to not upset anyone)

Comment (johanandren) [inline docs/src/main/paradox/connectors.md]: I sorted on contributor family name ;)

File: file/src/main/java/akka/stream/alpakka/file/javadsl/DirectoryChangesSource.java [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+  private final SourceShape<T> shape = SourceShape.of(out);
+
+  /**
+   * @param directoryPath Directory to watch
+   * @param pollInterval Interval between polls to the JDK watch service when a push comes in and there was no changes, if
+   *                     the JDK implementation is slow, it will not help lowering this
+   * @param maxBufferSize Maximum number of buffered directory changes before the stage fails
+   * @param combiner A function that combines a Path and a DirectoryChange into an element that will be emitted downstream
+   */
+  public DirectoryChangesSource(Path directoryPath, FiniteDuration pollInterval, int maxBufferSize, BiFunction<Path, DirectoryChange, T> combiner) {
```
Comment (ktoso) [inline file/src/main/java/akka/stream/alpakka/file/javadsl/DirectoryChangesSource.java]: good

**Comment by ktoso** [comment]
LGTM

**Comment by johanandren** [comment]
@2m fyi, will merge when validation completes

**Comment by 2m** [comment]
LGTM

**Pull request merged**